### PR TITLE
Ensure wrapped ERC20 transfers are correctly authorized and add end-to-end tests for them

### PR DESCRIPTION
**Pull request by james-chf**
Closes https://github.com/anoma/namada/issues/432 and relates to https://github.com/anoma/namada/issues/508

This PR
- ensures we check the relevant VPs are being executed for wrapped ERC20 transfers
- adds end-to-end tests covering
  - transfers of DAI from Ethereum to implicit and established Namada accounts
  - authorized transfers of wrapped DAI between implicit <-> established accounts once on Namada
  - rejecting transactions that attempt to transfer wrapped DAI from implicit/established accounts, if the transactions aren't signed by the key which controls the sending account

NB: some helper fns added/used in this PR are also in https://github.com/anoma/namada/pull/886/files#diff-b26bbd416d9d500f8f12aee8f7b3182f35b5cacb748898d9c07fe2cdd41e69b8

### CODE REVIEW

**Review by sug0 — APPROVED** [review]
maybe I'd change some of the `INFO` to `DEBUG` logs, but otherwise lgtm

File: shared/src/ledger/native_vp/ethereum_bridge/authorize.rs [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
-) -> Result<bool> {
-    tracing::warn!(
-        "authorize::is_authorized is not implemented, so all transfers are \
-         authorized"
-    );
-    Ok(true)
+    verifiers: &BTreeSet<Address>,
+    sender: &Address,
+) -> bool {
+    verifiers.contains(sender)
```
Comment (batconjurer) [inline shared/src/ledger/native_vp/ethereum_bridge/authorize.rs:14]: honest question, I've always wondered if we trigger the receiver vp as well. Or can people not refuse money from certain accounts?

**Review by batconjurer — COMMENTED** [review]
Generally looks good, I'm just waiting on an answer as to whether we trigger receiver vps for token transfers.

File: shared/src/ledger/native_vp/ethereum_bridge/authorize.rs (hunk 1 above)
Comment (batconjurer) [reply shared/src/ledger/native_vp/ethereum_bridge/authorize.rs:14]: It seems we should trigger the receiver account as well.

**Review by batconjurer — CHANGES_REQUESTED** [review]
Make sure the receiver vp also gets triggered.

File: shared/src/ledger/native_vp/ethereum_bridge/authorize.rs (hunk 1 above)
Comment (james-chf) [reply shared/src/ledger/native_vp/ethereum_bridge/authorize.rs:14]: Will add this, will require some small refactoring.

**Pull request merged**